src/net_peer.rs

```diff
@@ -192,25 +192,39 @@ impl Drop for MyTcpStream {
     }
 }
 
-pub fn tcp_connect_peer(addr: &SocketAddr) -> BoxedNewPeerFuture {
-    Box::new(
+pub fn resolve_addr(
+    addr_str: &str,
+) -> impl Future<Item = SocketAddr, Error = Box<dyn std::error::Error + 'static>> {
+    // TODO: resolve asynchronously
+    addr_str
+        .to_socket_addrs()
+        .and_then(|mut addrs| {
+            addrs
+                .next()
```

Comment:
No [happy eyeballs](https://en.wikipedia.org/wiki/Happy_Eyeballs)?

Reply:
That would be a welcome extension.